tests/entrypoint.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+USER_ID=${LOCAL_USER_ID:-1001}
+
+echo "Starting with UID : $USER_ID"
+useradd --shell /bin/bash -u $USER_ID -o -c "" -m user
+export HOME=/home/user
+
+exec /usr/local/bin/gosu user "$@"
```

Comment:
`gosu` is great!

If you ever want to simplify things without having to download/check gpg (and potentially run into licensing issues -> only an issue with public docker images) the good old `chroot` can do the job as well:

```
exec chroot --userspec=$USER_ID --skip-chdir "$@"
```

Reply:
Looks like the `chroot` in our base image doesn't support `--skip-chdir`. The base images that we use are all built on top of Jessie: https://hub.docker.com/_/python/

    root@10dbaa600623:/# chroot --skip-chdir
    chroot: unrecognized option '--skip-chdir'
    Try 'chroot --help' for more information.
    root@10dbaa600623:/# chroot --version
    chroot (GNU coreutils) 8.23

Any idea for a good workaround? I would love to get all the `gosu` install boilerplate out of the Dockerfile :D

Reply:
@beniwohli Is there any particular reason why you should base your images on the `jessie` release which was released in 2015 as opposed to `stretch` i.e. debian:9?

Reply:
Those are the official Python docker images, so I don't have control over it. The `python:3.7` image is built on Stretch, but I guess they don't want to risk unnecessary breakage by updating the images for older versions to Stretch.

Reply:
@beniwohli Ah I was too fast. So it's the python image that is based on debian:8. Have you considered using `python:3.6-stretch` or `python:2.7-stretch`?

```
$ docker run --rm -ti python:3.6-stretch chroot --help
Usage: chroot [OPTION] NEWROOT [COMMAND [ARG]...]
  or:  chroot OPTION
Run COMMAND with root directory set to NEWROOT.

  --groups=G_LIST        specify supplementary groups as g1,g2,..,gN
  --userspec=USER:GROUP  specify user and group (ID or name) to use
  --skip-chdir           do not change working directory to '/'
      --help     display this help and exit
      --version  output version information and exit

If no command is given, run '${SHELL} -i' (default: '/bin/sh -i').

GNU coreutils online help: <http://www.gnu.org/software/coreutils/>
Report chroot translation bugs to <http://translationproject.org/team/>
Full documentation at: <http://www.gnu.org/software/coreutils/chroot>
or available locally via: info '(coreutils) chroot invocation'
```

Reply:
Ugh, turns out there are no Stretch-based images for `pypy:2` and `pypy:3` :(